kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/SaslInspectionIT.java

```diff
@@ -257,38 +326,11 @@ private static void assertClientsCanAccessCluster(KafkaCluster cluster,
         }
     }
 
-    // broker has PLAIN enabled
-    // proxy has SCRAM enabled
-    // client handshakes with PLAIN
-    // => client should not complete authentication
-    @Test
-    void shouldNotAuthenticateWhenNoCommonMechanism(@SaslMechanism(value = "PLAIN", principals = {
-            @SaslMechanism.Principal(user = "alice", password = "alice-secret") }) KafkaCluster cluster,
-                                                    Topic topic) {
-        var config = buildProxyConfig("SCRAM-SHA-256", cluster);
-
-        try (var tester = kroxyliciousTester(config);
-                var producer = tester.producer(Map.of(
-                        CommonClientConfigs.CLIENT_ID_CONFIG, "PLAIN-producer",
-                        CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SASL_PLAINTEXT",
-                        SaslConfigs.SASL_MECHANISM, "PLAIN",
-                        SaslConfigs.SASL_JAAS_CONFIG, """
-                                        org.apache.kafka.common.security.plain.PlainLoginModule required
-                                            username="alice"
-                                            password="alice-secret";
-                                """))) {
-            assertThat(producer.send(new ProducerRecord<>(topic.name(), "my-key", "my-value")))
-                    .failsWithin(5, TimeUnit.SECONDS)
-                    .withThrowableOfType(ExecutionException.class)
-                    .withCauseExactlyInstanceOf(UnsupportedSaslMechanismException.class);
-        }
-    }
-
-    private static ConfigurationBuilder buildProxyConfig(String e1, KafkaCluster cluster) {
+    private static ConfigurationBuilder buildProxyConfig(String enabledSaslMech, KafkaCluster cluster) {
         NamedFilterDefinition saslInspection = new NamedFilterDefinitionBuilder(
                 SaslInspection.class.getName(),
                 SaslInspection.class.getName())
-                .withConfig("enabledMechanisms", Set.of(e1))
+                .withConfig("enableInsecureMechanisms", "PLAIN".equals(enabledSaslMech))
```

Comment:
a bit of a roundabout way to configure this, I think it's the only usage of the `enabledSaslMech` string.